src/mame/apple/apple2.cpp

```diff
@@ -937,6 +944,53 @@ void apple2_state::ay3600_data_ready_w(int state)
 	}
 }
 
+void apple2_state::tk10_data_write(u8 data)
+{
+		if (m_maincpu->total_cycles() < 25000)
+		{
+			return;
+		}
+
+		if (data & 0x80)
+		{
+			m_transchar = data & 0x7f;
+			m_strobe = 0x80;
+//      printf("new char = %04x (%02x)\n", m_lastchar&0x3f, m_transchar);
+		}
```

Comment:
If the top bit is the strobe, you need to actually look for edges.

Reply:
It seems to work fine without edge detection, I don't think the 8039 will write the OUTL BUS unless it actually changes, so it's won't write the OUTL BUS twice with the strobe high.

 0:147  mov  r6,a                                           AE
 0:148  jmp  $109                                           24 0
 0:14A  mov  a,r5                                           FD
 0:14B  orl  a,#$80                                         43 8
 0:14D  outl bus,a                                          02
 0:14E  mov  r5,#$10                                        BD 1
 0:150  djnz r5,$150                                        ED 5
 0:152  anl  a,#$7F                                         53 7
 0:154  outl bus,a                                          02

Reply:
You’re thinking in terms that are too narrow.  If it were to write two successive values with the strobe high, that isn’t going to cause the value to be latched twice.  Also, which edge is the host computer actually sensitive to?  The rising edge or the falling edge?